### PR TITLE
BUG: reset_index not preserving object dtype for string option

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/methods/test_reset_index.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def test_reset_index_inplace_and_drop_ignore_name(self):
         expected = Series(range(2), name="old")
         tm.assert_series_equal(ser, expected)
 
+    def test_reset_index_drop_infer_string(self):
+        # GH#56160
+        ser = Series(["a", "b", "c"], dtype=object)
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_reset_index.py:168]: This test doesn't need to involve the `infer_string` option?

Comment (phofl) [reply pandas/tests/series/methods/test_reset_index.py:170]: Oh yikes, added It, thx

**Comment by lithomas1** [comment]
pre-commit.ci autofix

**Review by lithomas1 — APPROVED** [review]
This LGTM on green.

**Comment by lithomas1** [comment]
thanks @phofl

**Pull request merged**